common/adapter.py

```diff
@@ -147,6 +147,10 @@ def _run_layer_train(
         logger.debug("Fetched input dataframe - {}", input_df.shape)
 
         # build the dataframe
+        layer_api_key = self.config.credentials.layer_api_key
+        if layer_api_key is None:
+            raise RuntimeException("Missing credentials: layer_api_key.")
```

Comment:
For better devex, should we use do `layer.login()` here? Otherwise, in the docs we will have to add a whole new section how to get the api key from layer and put into dbt profile. Wdyt?

Reply:
For `train/autoML`, we need to do login to be able to write the results.  (That's why it was not working)
For `predict`, we don't need the login unless they'd like to use their own private model -- that case we didn't consider yet.

Do you have another option in mind?

Reply:
I mean, we can check for the credentials in dbt profile, if it doesn't exist, we can ask user to login in the terminal just like in the notebook. Don't we show login url and box in the terminal if user not logged in?

I mean something like this:
```python
layer_api_key = self.config.credentials.layer_api_key
if layer_api_key is None:
   layer.login()
```

Reply:
I'm not sure what we do in the notebook but it doesn't work here.
![image](https://user-images.githubusercontent.com/874997/171912167-28be9a32-62a2-49af-a2d7-07f1b132d3ad.png)

Reply:
I have been thinking about it and I think that fail-fast should be the way to go here. 
The main `dbt` usage pipeline automation, so I think we should opt for an automation-friendly option.